cpu/stm32f3/periph/pwm.c

```diff
@@ -134,37 +142,44 @@ int pwm_set(pwm_t dev, int channel, unsigned int value)
 
     switch (dev) {
 #if PWM_0_EN
+
         case PWM_0:
             tim = PWM_0_DEV;
             break;
 #endif
 #if PWM_1_EN
+
         case PWM_1:
             tim = PWM_1_DEV;
             break;
 #endif
+
         default:
             return -1;
     }
 
-    /* norm value to maximum possible value */
-    if (value > 0xffff) {
-        value = 0xffff;
+    /* norm value to actual resolution */
+    if (value > tim->ARR) {
+        value = tim->ARR;
```

Comment:
If we go with the comment solution I'd try to find a synonym for "overriding" as is reads like "overwriting"  (or am I the only one reading it this way :-) ?). I don't mind which solution we take. @haukepetersen please state your opinion so we can merge this soon.